### PR TITLE
meta-raspberrypi: Enabling vc4graphics machine feature.

### PR DESCRIPTION
**Pull request by kalyan-nagabhirava**
gstreamer1.0-plugins-bad : Packaging gstreamer
plugins.

Signed-off-by: Riyaz <Riyaz.l@ltts.com>

### CODE REVIEW

**Comment by Khan3033** [comment]
> Please divide this into two separate patches. And I would like to gather more information on before switching default to vc4graphics on 32bit.

We have addressed your comments kindly review, thanks.

**Comment by kraj** [comment]
switching to 4.19 is being addressed in https://github.com/agherzan/meta-raspberrypi/pull/418
so make this pull just to swich the graphics driver, and we should make it across all rpis and just not for rpi3 alone.

File: conf/machine/raspberrypi3.conf [hunk 1]
```diff
@@ -13,6 +13,8 @@ MACHINE_EXTRA_RRECOMMENDS += "\
     bluez-firmware-rpidistro-bcm4345c0-hcd \
 "
 
+MACHINE_FEATURES += "vc4graphics"
```
Comment (kraj) [inline conf/machine/raspberrypi3.conf]: this change should be make common across all rpi machine, rest of the change should be dropped from this patch.

Comment (Khan3033) [reply conf/machine/raspberrypi3.conf]: Hi @kraj ,
Committed as per the change, Kindly review,
9cd3dc794461ccfb6c1354d16a8fc7deabe53c3d

**Comment by psivasubramanian** [comment]
Hi @kraj @agherzan 
We hope your review comments are addressed. Please verify and let us know if any comments.

**Comment by agherzan** [comment]
@psivasubramanian I'm not sure how mu comment is addressed. Maybe it was missed:

> The commit on this PR mentions nothing about why this is needed. What is the issue it addresses? Why do we want to enable by default vc4 for RaspberryPi 3? Is this part of a conversation I'm missing the context from? But even so, the summary of it should be available in the commit log.

This changes now a little by making the question non-rpi3 specific but the core of it still stands.

**Comment by agherzan** [comment]
I am tempted to consider this PR stale. I'll close it soon if I get no feedback.

**Comment by psivasubramanian** [comment]
> @psivasubramanian I'm not sure how mu comment is addressed. Maybe it was missed:
> 
> > The commit on this PR mentions nothing about why this is needed. What is the issue it addresses? Why do we want to enable by default vc4 for RaspberryPi 3? Is this part of a conversation I'm missing the context from? But even so, the summary of it should be available in the commit log.
> 
> This changes now a little by making the question non-rpi3 specific but the core of it still stands.

Sorry for the delayed reply. We thought the commit log (as below) answered your question to some extent. But it seems it doesn't.

meta-raspberrypi: Enabling open-source vc4graphics driver for all RPI platforms
Appending vc4graphics machine feature to use vc4graphics driver.

We are in the process of evaluating the performance of vc4graphics against userland for providing justification to this PR. Does that will be suffice?

**Comment by agherzan** [comment]
@psivasubramanian I've never received that message and the commit log just translates the commit content. There is not explanation whatsoever. Even so, you can always customize your build to switch on vc4. Isn't that working for you?

**Comment by kraj** [comment]
> @psivasubramanian I've never received that message and the commit log just translates the commit content. There is not explanation whatsoever. Even so, you can always customize your build to switch on vc4. Isn't that working for you?

@agherzan with linux kernel 4.19+ vv4graphics is in pretty good shape, so switching it to be default may be a compelling option for 32bit builds as well, we already use it by default for 64bit.

**Comment by psivasubramanian** [comment]
Hi @agherzan 
Hope this commit log (https://github.com/agherzan/meta-raspberrypi/pull/417/commits/680b89471dde76b5e9650e3865be8d01911faf69) gives more information about this change as well as answers your question. Kindly let us know if any comments/thoughts.

**Comment by agherzan** [comment]
@kraj We do but we will need some testing on 32B as well. It doesn't feel right to switch just like that. What about building the support but don't activating the overlay by default? My reason behind being a little reticent about this is because while working on RPi4 I had a hard time working with vc4 and finally managed to only make it work with fkms. CC @psivasubramanian

**Comment by kraj** [comment]
> @kraj We do but we will need some testing on 32B as well. It doesn't feel right to switch just like that. What about building the support but don't activating the overlay by default? My reason behind being a little reticent about this is because while working on RPi4 I had a hard time working with vc4 and finally managed to only make it work with fkms. CC @psivasubramanian

fkms issue was a regression in kernel which should be fixed with
https://github.com/agherzan/meta-raspberrypi/pull/436

So we should drop fkml for rpi4 as well after above is merged.

vc4 is where things are headed to, mesa works equally well on both 32bit and 64bit and
with 4.19 it also supports v4l2, so I think we are better off supporting it as default now a days
we can always revert or define another machine type something like
`rpi3-vc4graphics` and so on

**Comment by agherzan** [comment]
@kraj you are the main point of control in that area so I'll take your advice. Let's see who screams.

**Comment by agherzan** [comment]
@kraj can you also update your review, please?

**Comment by psivasubramanian** [comment]
@kraj @agherzan Rebased onto latest. Please review.

**Comment by agherzan** [comment]
@kraj @psivasubramanian I had to rebase master because the log of this commit was really ugly which I haven't caught at review this time.

**Pull request merged**